modules/serverless-troubleshooting-general.adoc

```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// * serverless/reference/troubleshooting.adoc
+
+:_content-type: REFERENCE
+[id="serverless-troubleshooting-general_{context}"]
+= General troubleshooting
+
+*How many {product-title} resources does {ServerlessProductShortName} Serving consume?*
+
+The request of the `knative-serving` component is 0.61 CPU and 400 megabytes of memory.
+
+The limit of the `knative-serving` component is 4.1 CPU and 4.71 gigabytes of memory.
```

Comment:
@mgencur Can you please provide the following:
* In `(300m + 100m + 100m + 30m + 100m) * 2`, what do the individual constituents stand for?
  * Or, if it is not important, can we just write `630m for every replica, resulting in 1260m for the default two replicas`?
* What are the numbers of Kourier-related pods?
  * Or should we just state `plus the requests of the Kourier-related pods`?
* What are the memory _limits_ for Knative Serving components and for Kourier-related pods?

Reply:
Here are some details. The CPU requests are defined for each Pod in knative-serving namespace.
```
oc get pods -n knative-serving
NAME                                     READY   STATUS    RESTARTS   AGE
activator-77d49fdbc-7hgr7                2/2     Running   0          11d
activator-77d49fdbc-8whqr                2/2     Running   0          11d
autoscaler-6444dcb6c9-489jd              2/2     Running   0          11d
autoscaler-6444dcb6c9-vrldn              2/2     Running   0          11d
autoscaler-hpa-59698cf457-2npx9          2/2     Running   0          11d
autoscaler-hpa-59698cf457-ntk5p          2/2     Running   0          11d
controller-54cf5b4ccc-6bb9x              2/2     Running   0          11d
controller-54cf5b4ccc-tht92              2/2     Running   0          11d
domain-mapping-7546dfd485-4xplg          2/2     Running   0          11d
domain-mapping-7546dfd485-hkrxs          2/2     Running   0          11d
domainmapping-webhook-64784df6d7-2295s   2/2     Running   0          11d
domainmapping-webhook-64784df6d7-wjcq4   2/2     Running   0          11d
webhook-7874c59b84-56pgs                 2/2     Running   0          11d
webhook-7874c59b84-lbfns                 2/2     Running   0          11d

oc get pod -n knative-serving -ojsonpath='{.items[*].spec.containers[*].resources.requests.cpu}'
300m 10m 300m 10m 100m 10m 100m 10m 30m 10m 30m 10m 100m 10m 100m 10m 30m 10m 30m 10m 100m 10m 100m 10m 100m 10m 100m 10m

oc get pod -n knative-serving -ojsonpath='{.items[*].spec.containers[*].resources.requests.memory}'
60Mi 20Mi 60Mi 20Mi 100Mi 20Mi 100Mi 20Mi 40Mi 20Mi 40Mi 20Mi 100Mi 20Mi 100Mi 20Mi 40Mi 20Mi 40Mi 20Mi 100Mi 20Mi 100Mi 20Mi 100Mi 20Mi 100Mi 20Mi
```
Overall, this is:
CPU request: 1660m
Memory request: 1360Mi

However, we don't define CPU and memory limits for all Pods, only for some of them. So it's probably doesn't make sense to mention CPU/memory limits because they can grow beyond the number you will get.

Also, the Pods in knative-serving-ingress namespace don't define any CPU/memory requests or limits. There's a JIRA for that: https://issues.redhat.com/browse/SRVKS-532
But there are some Pods:
```
oc get pods -n knative-serving-ingress
NAME                                      READY   STATUS    RESTARTS   AGE
3scale-kourier-gateway-785c74cd65-8qb2g   1/1     Running   0          11d
3scale-kourier-gateway-785c74cd65-s7w74   1/1     Running   0          11d
net-kourier-controller-7976dbcbbb-jdpdr   1/1     Running   0          11d
net-kourier-controller-7976dbcbbb-m5qfq   1/1     Running   0          11d
```
These Pods belong to "knative-serving" component. So, when talking about CPU/memory requests we may need to say "knative-serving" component requests "at least" XYZ memory and XYZ cpu.

Reply:
@mgencur Perfect, I've added this information, does it look good now?
https://deploy-preview-45262--osdocs.netlify.app/openshift-enterprise/latest/serverless/reference/troubleshooting.html#serverless-troubleshooting-general_serverless-troubleshooting
Thank you.

Reply:
Originally, my intention was to only explain where the numbers are coming from and not to include them in the docs in the given format. But it doesn't look too bad. This is probably as much info as we can provide.